docs/pages/cairo/enum.md

```diff
@@ -1,21 +1,69 @@
+```rust
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+enum PlayerCharacter {
+    Godzilla,
+    Dragon,
+    Fox,
+    Rhyno
+}
 
-You can define an enum as follows:
+```
+In this example, we've defined an enum called PlayerCharacter with four variants: `Godzilla`, `Dragon`, `Fox`, and `Rhyno`. The naming convention is to use PascalCase for enum variants. Each variant represents a distinct value of the PlayerCharacter type. In this particular example, variants don't have any associated value. 
+Now let's imagine that our variants have associated values, We can define a new PlayerCharacter enum:
 
 ```rust
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+enum PlayerCharacter {
+    Godzilla :u128,
+    Dragon : u128,
+    Fox : u128,
+    Rhyno : u128
+}
+
+```
+
+In the next example, we will see that it is also possible to associate different data types with each variant. The PlayerCharacter enum we will define serves as a blueprint for creating various types of characters, each with their unique attributes. This design choice is particularly useful in game development, where you often need to manage different types of entities (like characters, items, or enemies) that share some common properties but also have unique characteristics.
+
+``` rust 
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+enum PlayerCharacter {
+    Godzilla ,
+    Fox  : felt252,
+}
+
+    // Instantiate some characters with associated values
+let godzilla = PlayerCharacter::Godzilla;
+let fox = PlayerCharacter::Fox(90);
+```
 
-// This enum simply defines the states of a game.
-#[derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+In this example, the PlayerCharacter enum has three variants: `Godzilla`, `Fox`, and `Rhyno`, all with different types:
+`Godzilla` doesn't have any associated value.
+`Fox ` is a single felt252.
+`Rhyno` is a tuple of two u128 values.
+You could even use a Struct or another enum you defined inside one of your enum variants.
+
+## Trait Implementations for Enums
+
+ traits are a way to define shared behavior across types. By defining traits and implementing them for your custom enums, you can encapsulate common behaviors and methods that are relevant to the enum. This approach enhances code reusability and maintainability, especially in complex systems like game development.
+
+Consider the GameStatus enum, which represents the various states a game can be in. This enum is a simple yet powerful example of how enums can be used to model game states.
+
+ ``` rust
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
 enum GameStatus {
     NotStarted: (),
     Lobby: (),
     InProgress: (),
     Finished: (),
 }
+```
 
-// We define an into trait
+<!-- We define an into trait -->
```

Comment:
From where did you take the `<!-- -->`, html I guess? :)
In cairo, we use `//`. 👍